### PR TITLE
Add domain and nosto account in the API header

### LINKED ISSUE #494: Send active domain and Nosto account with API calls

### PR DESCRIPTION
**Pull request by olsiqose-nosto**
## Description
Add store domain and Nosto account in the headers of the API call. 
Cases added:
* Upsert Product
* Order Confirmation

## Related Issue
closes #494

## How Has This Been Tested?
Tested manually, checked the headers are correct when making the call.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have assigned the correct milestone or created one if non existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers

### CODE REVIEW

File: app/code/community/Nosto/Tagging/Helper/Url.php [hunk 1]
```diff
@@ -475,6 +475,23 @@ public function getFrontPageUrl(Mage_Core_Model_Store $store)
...
+     * @return string
+     */
+    public function getActiveDomain($store)
+    {
+        $storeFrontUrl = $this->getFrontPageUrl($store);
+        try {
+            return  Nosto_Helper_UrlHelper::parseDomain($storeFrontUrl);
+        } catch (\Exception $e) {
+            Nosto_Tagging_Helper_Log::exception($e);
+            return '';
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Helper/Url.php]: Should this be empty string or null?

Comment (olsiqose-nosto) [reply app/code/community/Nosto/Tagging/Helper/Url.php]: No main reason for empty string. It is used as default value in SDK UpserProduct() constructor. https://github.com/Nosto/nosto-php-sdk/blob/develop/src/Operation/UpsertProduct.php#L66

File: app/code/community/Nosto/Tagging/Helper/Url.php [hunk 2]
```diff
@@ -475,6 +475,23 @@ public function getFrontPageUrl(Mage_Core_Model_Store $store)
...
+     * Returns the store domain
+     *
+     * @param $store
+     * @return string
+     */
+    public function getActiveDomain($store)
+    {
+        $storeFrontUrl = $this->getFrontPageUrl($store);
+        try {
+            return  Nosto_Helper_UrlHelper::parseDomain($storeFrontUrl);
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Helper/Url.php]: Remove extra whitespace after `return`

File: app/code/community/Nosto/Tagging/Model/Service/Product.php [hunk 3]
```diff
@@ -110,7 +110,8 @@ protected function update(array $products)
             $env = $emulation->startEnvironmentEmulation($store->getId());
             foreach ($productBatches as $productsInStore) {
                 try {
-                    $operation = new Nosto_Operation_UpsertProduct($account);
+                    $urlHelper = Mage::helper('nosto_tagging/url');
+                    $operation = new Nosto_Operation_UpsertProduct($account, $urlHelper->getActiveDomain($store));
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Service/Product.php]: Perhaps move these outside the loop where we get all other helpers as well. Most likely it's not a costly call but there's no need to get the helper and the URL for each batch separately.

File: app/code/community/Nosto/Tagging/Model/Service/Product.php [hunk 4]
```diff
@@ -203,7 +204,8 @@ public function updateOutOfSyncToNosto()
                     );
                     break;
                 }
-                $operation = new Nosto_Operation_UpsertProduct($account);
+                $urlHelper = Mage::helper('nosto_tagging/url');
+                $operation = new Nosto_Operation_UpsertProduct($account, $urlHelper->getActiveDomain($store));
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Service/Product.php]: Same thing here, move outside loop and assign into a variable after the store is set (line 192 for example)

**Pull request merged**